gravitee-apim-console-webui/src/management/api/details/api-portal-details.component.quality.spec.ts

```diff
@@ -0,0 +1,158 @@
+import { User } from '../../../entities/user';
+import { Category } from '../../../entities/category/Category';
+import { QualityRule } from '../../../entities/qualityRule';
+
+describe('ApiPortalDetailsComponent - quality', () => {
+  const API_ID = 'apiId';
+  const currentUser = new User();
+  currentUser.userPermissions = ['api-definition-u'];
+  const fakeRootScope = { $broadcast: jest.fn(), $on: jest.fn() };
+
+  let fixture: ComponentFixture<ApiPortalDetailsComponent>;
+  let httpTestingController: HttpTestingController;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [NoopAnimationsModule, GioHttpTestingModule, ApiPortalDetailsModule, MatIconTestingModule],
+      providers: [
+        { provide: UIRouterStateParams, useValue: { apiId: API_ID } },
+        { provide: CurrentUserService, useValue: { currentUser } },
+        {
+          provide: 'Constants',
+          useValue: {
+            ...CONSTANTS_TESTING,
+            env: {
+              ...CONSTANTS_TESTING.env,
+              settings: {
+                ...CONSTANTS_TESTING.env.settings,
+                apiReview: {
+                  enabled: true,
+                },
+                apiQualityMetrics: {
+                  ...CONSTANTS_TESTING.env.settings.apiQualityMetrics,
+                  enabled: true,
+                },
+              },
+            },
+            org: {
+              ...CONSTANTS_TESTING.org,
+              settings: {
+                ...CONSTANTS_TESTING.org.settings,
+                jupiterMode: {
+                  enabled: true,
+                },
+              },
+            },
+          },
+        },
+        { provide: AjsRootScope, useValue: fakeRootScope },
+      ],
+    }).overrideProvider(InteractivityChecker, {
+      useValue: {
+        isFocusable: () => true, // This traps focus checks and so avoid warnings when dealing with
+      },
+    });
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ApiPortalDetailsComponent);
+
+    httpTestingController = TestBed.inject(HttpTestingController);
+    fixture.detectChanges();
+    trackImageOnload();
+  });
+
+  afterEach(() => {
+    httpTestingController.verify();
+  });
+
+  it('should display quality', async () => {
+    const api = fakeApi({
+      id: API_ID,
+      visibility: 'PRIVATE',
+    });
+
+    expectApiGetRequest(api);
+    expectCategoriesGetRequest();
+    expectQualityRulesGetRequest([{ id: 'ruleId', name: 'Custom rule', description: 'Custom rule', weight: 42 }]);
+    expectQualityMetricsGetRequest(API_ID, { score: 42, metrics_passed: { ruleId: true, 'api.quality.metrics.logo.weight': true } });
+    await waitImageCheck();
+    fixture.detectChanges();
+
+    const qualityMetrics: NodeList = fixture.nativeElement.querySelectorAll('.api-quality__content__list__item');
+
+    expect(qualityMetrics.length).toBe(2);
```

Comment:
same here code style so you can keep your code but most of the time I would have done something like the following
```suggestion
    expect(qualityMetrics).toHaveLength(2);
```